iiod/ops.c

```diff
@@ -560,7 +560,10 @@ static void rw_thd(struct thread_pool *pool, void *d)
 				struct iio_channel *chn = dev->channels[i];
 				long index = chn->index;
 
-				if (index >= 0 && TEST_BIT(entry->mask, i))
+				if (index < 0)
+					continue;
+
+				if (TEST_BIT(entry->mask, i))
```

Comment:
curios: should this be `chn->number` instead of `i`?

Reply:
The assumption is that both are the same! 
But Lars also thinks that chn->number looks more robust.
I'll fix this.